src/components/UI/BaseInput.vue

```diff
@@ -0,0 +1,54 @@
+<template>
+    <div class="base-input">
+        <label class="base-input__label" :for="$attrs.id">{{inputLabel}}</label>
+        <input class="base-input__input" v-bind="$attrs" v-on="inputListeners" :value="value" :class="[{'base-input__input--error': isInvalid}]"/>
+    </div>
+</template>
+
+<script>
+export default {
+    inheritAttrs: false,
+    props: {
+        value: {
+            type: String
+        },
+        inputLabel: {
+            type: String
+        },
+        isInvalid: {
+            type: Boolean
+        }
+    },
+    computed: {
+        inputListeners() {
+            const vm = this
+            return Object.assign({},
+                vm.$listeners,
+                {
+                    input: function(event) {
+                        vm.$emit('input', event.target.value)
+                    },
+                    blur: function(event) {
+                        vm.$emit('blur', event.target.value)
+                    }
```

Comment:
Until I found this, I struggled a lot to create a transparent component. 
Now im able to extract functionalities as events and inject in right place.
I recommend it without hesitation 👍

Reply:
very nice finding with this `v-on="..."`  👏 

Two tiny hints:

```js
{
  input: function(event) { vm.$emit('input', event.target.value) }
}
```
is equivalent to
```js
{
  input(event) { vm.$emit('input', event.target.value) }
}
```
imo the later is a bit cleaner

Secondly, instead of using `Object.assign` you can do it a bit shorter:
```js
inputListeners() {
  return {
    ...this.$listeners,
    input: (event) => this..$emit('input', event.target.value),
    blur: (event) => this.$emit('blur', event.target.value),
  }
}
```
which should have the same effect 😉

Reply:
That's what vue documentation says but you are so right 😌

Reply:
Vue 2 docs in most parts were written before ES6 was a standard so some parts are a bit clunky 😉

Reply:
That make sense, got it! 📝